### ISSUE TITLE
[Java] Bump org.apache.maven dependencies from 3.8.7 to 3.9.6

### ISSUE THREAD

**Issue by ianmcook** [apache/arrow#40714]
### Describe the enhancement requested

After #40514 merges (closing #40515), let's try again to bump Maven to 3.9.6.

We should maybe open a new PR that supersedes #39451 because rebasing that might be messy.

### Component(s)

Java

**Comment by vibhatha** [comment]
take

**Comment by vibhatha** [comment]
Sorry @ianmcook I didn't see you have self assigned. Please ignore my take, I checked from github mobile 🙏

**Comment by ianmcook** [comment]
@vibhatha you're welcome to take it :)
